extensions/reg-core-3.3.md

```diff
@@ -0,0 +1,261 @@
+---
+title: "The REG command framework"
+layout: spec
+copyrights:
+  -
+    name: "William Pitcock"
+    period: "2014-2015"
+    email: "nenolod@dereferenced.org"
+  -
+    name: "Daniel Oaks"
+    period: "2016"
+    email: "daniel@danieloaks.net"
+---
+
+The `REG` command framework provides a unified, standardized interface for account creation and
+validation to the network's authentication layer.  This allows for a network to provide a common
+interface regardless of what authentication layer they have chosen for their network to operate,
+i.e. traditional services, a different central authority, or a decentralized model similar to [IRCX][ircx].
+
+   [ircx]: http://en.wikipedia.org/wiki/IRCX
+   
+We believe that the benefit of adding a common interface for account creation and validation,
+is helpful to the end user in discovering the capabilities offered by the authentication and
+authority layer of the network.  Further, we believe it is helpful to client authors, so that
+they may provide guided account creation and validation for their users, regardless of network
+configuration.
+
+## The `REG` command
+
+The `REG` command is the main command used by the account registration subsystem.  It provides
+several subcommands listed in the `REGCOMMANDS` ISUPPORT type.  The core sub-commands are listed
+below:
+
+## The `REG CREATE` sub-command
+
+The `REG CREATE` command signals the intent of a client to register an account in the network's
+authentication layer.  It is similar to current methods of signalling that intent.
+
+A `REG CREATE` command consists of the following format:
+
+    REG CREATE <accountname> [callback_namespace:]<callback> [cred_type] :<credential>
+
+The `accountname` field specifies the account name the client wishes to register.  If the account name
+is `*`, it indicates that the client wishes to register with their current nickname as this field.
+
+The `credential` field specifies a credential of `cred_type`.  If there is no specified credential, then
+the `credential` field MUST be `*`.  A passphrase type `credential` MAY have spaces in it. New credential
+types may or may not allow spaces inside `credential`.
+
+The `callback` field designates an opaque value which indicates where a verification code, if any,
+will be sent.  The callback field is implementation-specific, with namespaces being listed using the
```

Comment:
If we're using IANA-registered scheme names for the callback namespace, does it make sense that the callback field is implementation-specific? Why not just use a proper URI here?